### PR TITLE
Add docs for A. O. Smith integration

### PR DESCRIPTION
**Pull request by bdr99**
## Proposed change

Add documentation for the A. O. Smith integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:
  - https://github.com/home-assistant/core/pull/104976
  - https://github.com/home-assistant/core/pull/105604
  - https://github.com/home-assistant/core/pull/105616
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/4947
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/aosmith.markdown [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  - Water heater
+ha_release: 2024.1
+ha_iot_class: Cloud Polling
+ha_domain: aosmith
+ha_config_flow: true
+ha_codeowners:
+  - '@bdr99'
+ha_platforms:
+  - water_heater
+ha_quality_scale: platinum
```
Comment (joostlek) [inline source/_integrations/aosmith.markdown]: Not yet ;)

**Review by joostlek — DISMISSED** [review]

File: source/_integrations/aosmith.markdown (hunk 1 above)
Comment (bdr99) [reply source/_integrations/aosmith.markdown]: It was platinum before you made me remove reauth from the PR 🙂

Comment (bdr99) [reply source/_integrations/aosmith.markdown]: Reauth PR: https://github.com/home-assistant/core/pull/105320

Comment (bdr99) [reply source/_integrations/aosmith.markdown]: Reauth is merged now. Can you please take another look at this PR @joostlek?

**Comment by bdr99** [comment]
Added the sensor platform, implemented in https://github.com/home-assistant/core/pull/105604.

File: source/_integrations/aosmith.markdown [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ha_integration_type: integration
+---
+
+The A. O. Smith integration allows you to control a compatible A. O. Smith iCOMM-enabled water heater from Home Assistant.
+
+Before using this integration, your water heater must be connected to a Wi-Fi network and linked to your account using the A. O. Smith mobile app ([iOS](https://apps.apple.com/us/app/a-o-smith/id456489822)/[Android](https://play.google.com/store/apps/details?id=com.aosmith.warrantycheck)).
+
+## Known Compatible Models
```
Comment (c0ffeeca7) [inline source/_integrations/aosmith.markdown]:
```suggestion
## Known compatible models
```

File: source/_integrations/aosmith.markdown (hunk 1 above)
Comment (joostlek) [reply source/_integrations/aosmith.markdown]:
```suggestion
```
Let's have a separate PR to bump it to platinum so the others have time and space to check if this is the case

Comment (bdr99) [reply source/_integrations/aosmith.markdown]: Removed.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @bdr99 :+1:

**Pull request merged**